### PR TITLE
Drive Cycle in Experiment

### PR DESCRIPTION
**Pull request by alibh95**
# Description

Added the functionality to simulate drive cycle using the experiment class.

Fixes # (issue)
#1193 
#1265 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/master/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [✓] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [✓] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [✓] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/experiments/experiment.py [hunk 1]
```diff
@@ -134,9 +143,18 @@ def read_string(self, cond):
             electric = self.convert_electric(cond_list[:idx])
             time = None
             events = self.convert_electric(cond_list[idx + 1 :])
+        elif "Run" in cond:
+            # e.g. Run US06
+            cond_list = cond.split()
+            electric = (drive_cycles[cond_list[1]][:, 1], "Drive")
+            # Set time and period to 1 second
+            # because drive cycles are sampled at 1Hz frequency
```
Comment (tinosulzer) [inline pybamm/experiments/experiment.py:151]: This is not always true. For drive cycles, the time and period should be set to "None" and then the difference between times in the drive cycle itself should be used.

File: pybamm/simulation.py [hunk 2]
```diff
@@ -211,39 +211,84 @@ def set_up_experiment(self, model, experiment):
...
+                    driving_inputs = {
+                        "Current switch": 1,
+                        "Voltage switch": 0,
+                        "Power switch": 0,
+                        "Current input [A]": I_drive,
+                        "Voltage input [V]": 0,  # doesn't matter
+                        "Power input [W]": 0,  # doesn't matter
+                    }
+                    # Update period
+                    driving_inputs["period"] = op[3]
```
Comment (tinosulzer) [inline pybamm/simulation.py:229]: here period should also be a list, which is updated based on the times in the drive cycle

File: pybamm/simulation.py [hunk 3]
```diff
@@ -417,38 +462,69 @@ def solve(
             self._solution = None
             # Step through all experimental conditions
             inputs = inputs or {}
-            pybamm.logger.info("Start running experiment")
```
Comment (tinosulzer) [inline pybamm/simulation.py:420]: why was this removed?

**Review by tinosulzer — CHANGES_REQUESTED** [review]
@alibh95 , thanks very much for taking this on! 

Can you add tests for the changes you have made?
After we merge #1278 , there will be a way to do this in a more efficient way, but this is a good first step.

File: pybamm/simulation.py (hunk 3 above)
Comment (alibh95) [reply pybamm/simulation.py:420]: It was removed by mistake. I will add this line in next commit.

**Comment by tinosulzer** [comment]
@alibh95 I have an idea for how to do this slightly differently. Are you still working on it?

**Comment by alibh95** [comment]
@tinosulzer Yes, I am working on this issue. Till now I am able to add the functionality of "for event" but unable to add the functionality of "until event".
